modules/engine/src/lib/pipeline-factory.ts

```diff
@@ -103,15 +103,15 @@ export class PipelineFactory {
     const bufferLayoutHash = this._getHash(JSON.stringify(props.bufferLayout));
 
     switch (this.device.info.type) {
-      case 'webgpu':
+      // case 'webgl':
+      // WebGL is more dynamic
+      // return `${vsHash}/${fsHash}V${varyingHash}BL${bufferLayoutHash}`;
+      default:
```

Comment:
Enabling this path fixes the issue on WebGL. Most other changes are to get the test cases working.

Reply:
It does feel better having the same path for both renderers here, and likely to help flush out bugs we might hit with WebGPU implementation later. 👍🏻